transaction_decoder/transaction_decoder_test.py

```diff
@@ -0,0 +1,112 @@
+from transaction_decoder.transaction_decoder import TransactionDecoder, TransactionToDecode
+
+
+class TestTransactionDecoder:
+    def test_nft_smart_contract_call(self) -> None:
+        tx_to_decode = TransactionToDecode()
+        tx_to_decode.sender = (
+            "erd18w6yj09l9jwlpj5cjqq9eccfgulkympv7d4rj6vq4u49j8fpwzwsvx7e85"
+        )
+        tx_to_decode.receiver = (
+            "erd18w6yj09l9jwlpj5cjqq9eccfgulkympv7d4rj6vq4u49j8fpwzwsvx7e85"
+        )
+        tx_to_decode.value = "0"
+        tx_to_decode.data = "RVNEVE5GVFRyYW5zZmVyQDRjNGI0ZDQ1NTgyZDYxNjE2MjM5MzEzMEAyZmI0ZTlAZTQwZjE2OTk3MTY1NWU2YmIwNGNAMDAwMDAwMDAwMDAwMDAwMDA1MDBkZjNiZWJlMWFmYTEwYzQwOTI1ZTgzM2MxNGE0NjBlMTBhODQ5ZjUwYTQ2OEA3Mzc3NjE3MDVmNmM2YjZkNjU3ODVmNzQ2ZjVmNjU2NzZjNjRAMGIzNzdmMjYxYzNjNzE5MUA="
+
+        decoder = TransactionDecoder()
+        metadata = decoder.getTransactionMetadata(tx_to_decode)
+
+        assert (
+            metadata.sender
+            == "erd18w6yj09l9jwlpj5cjqq9eccfgulkympv7d4rj6vq4u49j8fpwzwsvx7e85"
+        )
+        assert (
+            metadata.receiver
+            == "erd1qqqqqqqqqqqqqpgqmua7hcd05yxypyj7sv7pffrquy9gf86s535qxct34s"
+        )
+        assert metadata.value == 1076977887712805212893260
+        assert metadata.function_name == "swap_lkmex_to_egld"
+        assert metadata.function_args == ["0b377f261c3c7191", ""]
+        if metadata.transfers:
+            assert metadata.transfers[0].value == 1076977887712805212893260
```

Comment:
so the value will be a bignumber, right? there's no way that it will be interpreted as int, right?